xtask/src/codegen/gen_unstable_future_descriptor.rs

```diff
@@ -0,0 +1,62 @@
+//! Generates descriptors structure for unstable feature from Unstable Book
+
+use crate::codegen::update;
+use crate::codegen::{self, project_root, Mode, Result};
+use quote::quote;
+use std::fs;
+use std::process::Command;
+use walkdir::WalkDir;
+
+pub fn generate_unstable_future_descriptor(mode: Mode) -> Result<()> {
+    let path = project_root().join(codegen::STORAGE);
+    fs::create_dir_all(path.clone())?;
+
+    Command::new("git").current_dir(path.clone()).arg("init").output()?;
+    Command::new("git")
+        .current_dir(path.clone())
+        .args(&["remote", "add", "-f", "origin", codegen::REPOSITORY_URL])
+        .output()?;
+    Command::new("git")
+        .current_dir(path.clone())
+        .args(&["sparse-checkout", "set", "/src/doc/unstable-book/src/"])
+        .output()?;
+    Command::new("git").current_dir(path.clone()).args(&["pull", "origin", "master"]).output()?;
```

Comment:
We have `not_bash` module for this kind of scripting:

https://github.com/rust-analyzer/rust-analyzer/blob/master/xtask/src/not_bash.rs

```
let _d = pushd(path);
run!("git pull origin master")?
...
```